src/stdio_mgr/stdio_mgr.py

```diff
@@ -181,41 +222,35 @@ def stdio_mgr(in_str=""):
 
     out_
 
-        :cls:`~io.StringIO` -- Temporary stream for `stdout`,
+        :cls:`~io.RandomTextIO` -- Temporary stream for `stdout`,
         initially empty.
 
     err_
 
-        :cls:`~io.StringIO` -- Temporary stream for `stderr`,
+        :cls:`~io.RandomTextIO` -- Temporary stream for `stderr`,
         initially empty.
 
     """
     old_stdin = sys.stdin
     old_stdout = sys.stdout
     old_stderr = sys.stderr
 
-    new_stdout = StringIO()
-    new_stderr = StringIO()
-    new_stdin = TeeStdin(new_stdout, in_str)
+    with ExitStack() as stack:
+        new_stdout = stack.enter_context(RandomTextIO())
```

Comment:
Never even thought of using the already-existing context capabilities of these objects for better management of stream closure. Cool!